### PR TITLE
doc: update ChangeTable docstrings

### LINKED ISSUE #445: Improve documentation of element additions via the change table
The current `ChangeTable` docstring is out of date for storage (it still refers to passing a single dict, rather than a list of dicts), and the docstring is pretty vague for `add_branch`, `add_dcline`, `add_plant`, and `add_storage_capacity`.

### PR DESCRIPTION
**Pull request by danielolsen**
### Purpose
Add details to the docstrings of the ChangeTable methods for adding new elements, and update the main ChangeTable docstring to more accurately reflect the new storage schema (clean-up after https://github.com/Breakthrough-Energy/PowerSimData/pull/383).
Closes #445.

### What the code is doing
No code, just docstrings.

### Time estimate
5-10 minutes.

### CODE REVIEW

File: powersimdata/input/change_table.py [hunk 1]
```diff
@@ -520,6 +539,13 @@ def add_dcline(self, info):
 
         :param list info: each entry is a dictionary. The dictionary gathers
             the information needed to create a new dcline.
+            Required keys: "from_bus_id", "to_bus_id".
+            Optional keys: "capacity", "Pmax", "Pmin".
+            "capacity" denotes a bidirectional power limit (MW).
+            "Pmax" denotes a limit on power flowing from 'to' end to 'from' end.
```
Comment (BainanXia) [inline powersimdata/input/change_table.py:545]: "from 'from' end to 'to' end"?

Comment (danielolsen) [reply powersimdata/input/change_table.py]: Good catch, fixed.

File: powersimdata/input/change_table.py [hunk 2]
```diff
@@ -91,8 +91,11 @@ class ChangeTable(object):
         the and the scaling factor for the increase/decrease in capacity
         of the line as value.
     * *'storage'*:
-        value is a dictionary. The latter has *'bus_id'* as keys and the
-        capacity of storage (in MW) to add as value.
+        value is a list. Each entry in this list is a dictionary enclosing all the
+        information needed to add a new storage device to the grid. The keys in the
+        dictionary are: *'bus_id'*, *'capacity'*, "duration", "min_stor", "max_stor",
+        "energy_value", "InEff", "OutEff", "LossFactor", "terminal_min",
+        and "terminal_max". See the `add_storage_capacity` method for more details.
```
Comment (rouille) [inline powersimdata/input/change_table.py:98]: See the :meth:`add_storage_capacity` method

File: powersimdata/input/change_table.py [hunk 3]
```diff
@@ -646,6 +673,11 @@ def add_plant(self, info):
 
         :param list info: each entry is a dictionary. The dictionary gathers
             the information needed to create a new generator.
+            Required keys: "bus_id", "Pmax", "type".
+            Optional keys: "c0", "c1", "c2", "Pmin".
+            "c0", "c1", and "c2" are the coefficients for the cost curve, representing
+            the fixed cost ($/hour), linear cost ($/MWh), and quadratic cost ($/MW^2*h).
```
Comment (rouille) [inline powersimdata/input/change_table.py:679]: :math:\`\rm{MW}^2 \rm{h}\`

File: powersimdata/input/change_table.py (hunk 2 above)
Comment (danielolsen) [reply powersimdata/input/change_table.py]: Done.

File: powersimdata/input/change_table.py [hunk 4]
```diff
@@ -646,6 +673,11 @@ def add_plant(self, info):
...
+            These are optional for hydro, solar, and wind, and required for other types.
         :raises TypeError: if info is not a list.
```
Comment (rouille) [inline powersimdata/input/change_table.py:681]: info --> \`\`info\`\`

File: powersimdata/input/change_table.py [hunk 5]
```diff
@@ -467,6 +470,22 @@ def add_storage_capacity(self, info):
...
+            "energy_value" denotes the value of stored energy at interval end ($/MWh).
+            "InEff" denotes the input efficiency (unitless), e.g. 0.95 = 95%.
+            "OutEff" denotes the output efficiency (unitless), e.g. 0.95 = 95%.
+            "LossFactor" denotes the per-hour relative losses,
+            e.g. 0.01 means that 1% of the current state of charge is lost per hour).
+            "terminal_min" denotes the minimum state of charge at interval end,
+            e.g. 0.5 means that the storage must end the interval with at least 50%.
+            "terminal_max" denotes the maximum state of charge at interval end,
+            e.g. 0.9 means that the storage must end the interval with no more than 90%.
         :raises TypeError: if info is not a list.
```
Comment (rouille) [inline powersimdata/input/change_table.py:489]: info --> \`\`info\`\`

File: powersimdata/input/change_table.py (hunk 4 above)
Comment (danielolsen) [reply powersimdata/input/change_table.py:681]: With one ``` or two on each side?

File: powersimdata/input/change_table.py (hunk 3 above)
Comment (danielolsen) [reply powersimdata/input/change_table.py:679]: Do you want formatting on the quadratic term only, or all of them?

File: powersimdata/input/change_table.py (hunk 4 above)
Comment (rouille) [reply powersimdata/input/change_table.py:681]: two on each side for verbatim (it displays like `that`) in ReST. I started to reference to parameters in the docstring this way as we do in Slack or Markdown

Comment (BainanXia) [reply powersimdata/input/change_table.py:681]: @rouille Good to know.

File: powersimdata/input/change_table.py (hunk 3 above)
Comment (rouille) [reply powersimdata/input/change_table.py:679]: Will it change for the other ones? the exponent in the quadratic is the on;y thing that will be in math mode

Comment (BainanXia) [reply powersimdata/input/change_table.py:679]: First time see this representation. This is the classic way to represent the unit of the quadratic term?

Comment (danielolsen) [reply powersimdata/input/change_table.py:679]: @rouille: Ah, I get it now.

Comment (rouille) [reply powersimdata/input/change_table.py:679]: :math: in ReST lets you right some LaTeX math

Comment (danielolsen) [reply powersimdata/input/change_table.py:679]: @BainanXia I'm not sure. I think removing the `*` per @rouille's suggestion will make it clear. The units line up properly.

Comment (danielolsen) [reply powersimdata/input/change_table.py]: Done.

File: powersimdata/input/change_table.py (hunk 4 above)
Comment (danielolsen) [reply powersimdata/input/change_table.py]: Done.

File: powersimdata/input/change_table.py (hunk 5 above)
Comment (danielolsen) [reply powersimdata/input/change_table.py]: Done.

**Review by rouille — APPROVED** [review]
Thanks

**Review by BainanXia — APPROVED** [review]
:pray:

**Pull request merged**